### ISSUE TITLE
Hostname and Domain customization Ubuntu 20.04 not working

### ISSUE THREAD

**Issue by joaofeteira** [hashicorp/terraform-provider-vsphere#1462]
### Terraform Version
v0.15.4

### vSphere Provider Version
v2.0.2

### Affected Resource(s)
- `vsphere_virtual_machine`

### Terraform Configuration Files

Relevant configs:
variables.tf
```hcl
variable "vm-domain" {
    type = string
}

variable "vm-prefix" {
    type = string
}
```

terraform.tfvars
```hcl
vm-domain = "lab.k8s.test"
vm-prefix = "k8s"
```

main.tf
```hcl
    clone {
        template_uuid = data.vsphere_virtual_machine.template.id
        customize {
            timeout = 0
            
            linux_options {
            host_name = "${var.vm-prefix}-ctrl-${count.index + 1}"
            domain = "${var.vm-domain}"
            }
```

### Debug Output

https://gist.github.com/joaofeteira/404a6ca2d561180f8ff46041d82727e5

### Expected Behavior

VM set with hostname k8s-ctrl-1.lab.k8s.test

### Actual Behavior

VM set with hostname k8s-ctrl-1

### Steps to Reproduce

1. `terraform apply`

Thanks a lot

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by Calvinaud** [comment]
Hello,

I don't think this a bug.

The domain is not part of the *hostname* but of the *FQDN*. (You can check it with: ```hostname --fqdn```). 
It's also describe as part of the *FQDN* in the documentation of the vsphere provider <https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/resources/virtual_machine#domain>.

This links explain the difference between the hostname/domain/FQDN:  <https://serverfault.com/a/269840>

If you really want to have the hostname as your expected behaviour, you will need to use your "vm-domain" variable in the "hostname" section.

```hcl 
host_name = "${var.vm-prefix}-ctrl-${count.index + 1}.${var.vm-domain}"
```

 **But**, it's considered a really bad practice to use dots in a hostname since dots are used as delimiter by the DNS.

**Comment by tenthirtyam** [comment]
I agree with @Calvinaud . In fact, this would not even be valid if you were deploying a virtual machine the traditional way as - the customization specification would fail.

<img width="1095" alt="image" src="https://user-images.githubusercontent.com/7771363/133694699-6c708194-d9f3-419d-bd67-b9b9ebe57c95.png">

Ryan

**Comment by tenthirtyam** [comment]
Interestingly enough, I tested with:

- Terraform Version: v1.0.6
- vSphere Provider Version: v2.0.2

... and the name was blocked just like the UI from my previous comment.

```
  clone {
    template_uuid = data.vsphere_virtual_machine.ubuntu.id
      customize {
        timeout = 0        
        linux_options {
          host_name = "k8s-ctrl-1.lab.k8s.test"
          domain = "lab.k8s.test"
        }
        network_interface {}
      }
    }
```

Result:

```
Plan: 1 to add, 0 to change, 0 to destroy.
vsphere_virtual_machine.learn: Creating...
vsphere_virtual_machine.learn: Still creating... [10s elapsed]
╷
│ Error: error sending customization spec: A specified parameter was not correct: spec.identity.hostName
```

Ryan

**Comment by tenthirtyam** [comment]
Hi @joaofeteira - checking in to see if you are still having this issue or if the issue can be closed. Please review prior comments in the thread.

Thanks!
Ryan

**Comment by joaofeteira** [comment]
Hi guys,

you are completely right. My mistake. Sorry to keep you waiting on this. 
Regards

**Comment by tenthirtyam** [comment]
Thank you for updating this issue, @joaofeteira.
